### PR TITLE
Update gradle for github actions, add initial workflows

### PR DESCRIPTION
**Pull request by XiXiaPdx**
I've tested these changes locally and also published a "release" to sonatype. Verified the repo and signed artifacts for both projects do close, so they can be released. 

Copy:

Took the actions that already work in other open source java agent repos and copied them here

Notable:

The version string is created differently now.  I decided to make this align with the other repos using github actions.  By default, the version string ends in `SNAPSHOT`.  If gradlew passes `-Prelease=true`, will trigger the publish workflow (I will add this workflow after this PR is approved).  

So versioning will be like this:

0.7.0-SNAPSHOT  will be the default version as features are added
0.7.0 would be the Release of the snapshot above

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -16,7 +16,10 @@ apply(plugin = "com.github.sherter.google-java-format")
 
 allprojects {
     group = "com.newrelic.telemetry"
-    version = project.findProperty("releaseVersion") as String
+    // -Prelease=true will render a non-snapshot version
+// All other values (including unset) will render a snapshot version.
+    val release: String? by project
+    version = "0.6.1" + if("true" == release) "" else "-SNAPSHOT"
```
Comment (jkwatson) [inline build.gradle.kts:22]: this version feels very hidden down in the guts of the gradle script. Can we make it more prominent and obvious so it's very clear where to find it when we want to change it?

Comment (XiXiaPdx) [reply build.gradle.kts:22]: Thanks for calling that out. Would this larger comment suffice or would you prefer a method call here?

```suggestion
/* Setting version for release or snapshot
     *
     * Passing in -Prelease=true will render a non-snapshot version
     *
     * All other values (including unset) will render a snapshot version.
     */
     
    version = "0.6.1" + if("true" == release) "" else "-SNAPSHOT"
```

Comment (jkwatson) [reply build.gradle.kts:22]: Can we make the version a property at the top (or near the top) of the build script, or some such? I don't like having the only reference embedded in a line of kotlin that makes it harder to see.

Comment (XiXiaPdx) [reply build.gradle.kts:22]: I'll take a look

I tried moving it to the top, out of the `allprojects {}` closure, but when the script is resolved, each individual project received `unspecified` as their version.

Comment (jkwatson) [reply build.gradle.kts:22]: can you at least move it to the top of the `allProjects` with a name like `baseVersion`?

Comment (XiXiaPdx) [reply build.gradle.kts]: Ok, it was more obvious than I thought.   It's basically how it was before.  Just change the version in the gradle.properties

**Comment by jkwatson** [comment]
Can you also add a comment in the gradle.properties that explains how it's used? I think most people (myself included) will assume that the version in there is the "real" version, and not just a base version that might have SNAPSHOT appended to it.

**Comment by XiXiaPdx** [comment]
@jkwatson   Hmmm, do you know why this PR build is just stuck here?

**Comment by jkwatson** [comment]
it's probably still a requirement in the project to build on Azure...looking....

**Comment by jkwatson** [comment]
yep, we were still requiring the Azure pipelines build for PRs.

**Comment by XiXiaPdx** [comment]
I don't know where to go to turn Azure off ... I don't see azure anywhere in the repo anymore.  I also tried logging in and do not appear to have access anymore.  Help?

**Comment by jkwatson** [comment]
I turned it off.  :)

**Pull request merged**